store/storetest/thread_store.go

```diff
@@ -773,61 +817,206 @@ func testGetThreadsForUser(t *testing.T, ss store.Store) {
+				team1channel1post2, team1channel1post3, gm1post1, // (no deleted threads in team1)
+			}},
+			{"team2, user1", user1ID, team2.Id, model.GetUserThreadsOpts{}, []*model.Post{
+				team2channel1post1, dm1post1, gm1post1,
+			}},
+			{"team2, user1, unread", user1ID, team2.Id, model.GetUserThreadsOpts{Unread: true}, []*model.Post{
+				gm1post1, // (no unread in team2)
+			}},
+			{"team2, user1, deleted", user1ID, team2.Id, model.GetUserThreadsOpts{Deleted: true}, []*model.Post{
+				team2channel1post1, team2channel1post2deleted, dm1post1, gm1post1,
+			}},
+			{"team2, user1, unread + deleted", user1ID, team2.Id, model.GetUserThreadsOpts{Unread: true, Deleted: true}, []*model.Post{
+				team2channel1post2deleted, gm1post1,
+			}},
+		}
+
+		for _, testCase := range testCases {
+			t.Run(testCase.Description, func(t *testing.T) {
+				totalThreads, err := ss.Thread().GetTotalThreads(testCase.UserID, testCase.TeamID, testCase.Options)
+				require.NoError(t, err)
 
-		// 2 threads from team1, 1 threads from team2, 1 dm thread, 1 gm thread
-		assert.EqualValues(t, 5, threads.Total)
-		assert.EqualValues(t, 5, threads.TotalUnreadThreads)
-		assert.EqualValues(t, 0, threads.TotalUnreadMentions)
-		assert.Len(t, threads.Threads, 5)
+				assert.EqualValues(t, int64(len(testCase.ExpectedThreads)), totalThreads)
+			})
+		}
 	})
 
-	t.Run("team1 specified, user1", func(t *testing.T) {
-		threads, err := ss.Thread().GetThreadsForUser(user1ID, team1.Id, model.GetUserThreadsOpts{})
-		require.NoError(t, err)
+	t.Run("GetTotalUnreadMentions", func(t *testing.T) {
+		testCases := []struct {
+			Description string
+			UserID      string
+			TeamID      string
+			Options     model.GetUserThreadsOpts
+
+			ExpectedThreads []*model.Post
+		}{
+			{"all teams, user1", user1ID, "", model.GetUserThreadsOpts{}, []*model.Post{
+				team1channel1post3, gm1post1,
+			}},
+			{"team1, user1", user1ID, team1.Id, model.GetUserThreadsOpts{}, []*model.Post{
+				team1channel1post3, gm1post1,
+			}},
+			{"team2, user1", user1ID, team2.Id, model.GetUserThreadsOpts{}, []*model.Post{
+				gm1post1,
+			}},
+		}
+
+		for _, testCase := range testCases {
+			t.Run(testCase.Description, func(t *testing.T) {
+				totalUnreadMentions, err := ss.Thread().GetTotalUnreadMentions(testCase.UserID, testCase.TeamID, testCase.Options)
+				require.NoError(t, err)
 
-		// 2 threads from team1, 1 dm thread, 1 gm thread
-		assert.EqualValues(t, 4, threads.Total)
-		assert.EqualValues(t, 4, threads.TotalUnreadThreads)
-		assert.EqualValues(t, 0, threads.TotalUnreadMentions)
-		assert.Len(t, threads.Threads, 4)
+				assert.EqualValues(t, int64(len(testCase.ExpectedThreads)), totalUnreadMentions)
+			})
+		}
 	})
 
-	t.Run("team2 specified, user1", func(t *testing.T) {
-		threads, err := ss.Thread().GetThreadsForUser(user1ID, team2.Id, model.GetUserThreadsOpts{})
-		require.NoError(t, err)
+	t.Run("GetThreadsForUser", func(t *testing.T) {
```

Comment:
This isn't 100% coverage yet, but broader than before. (I know we have the `api4` tests already, but I found it helpful to test holistically here too.)